src/wrap_fn_ptr.rs

```diff
@@ -56,11 +43,29 @@ macro_rules! wrap_fn_ptr {
             $($arg_name:ident: $arg_ty:ty),*$(,)?
     ) -> $return_ty:ty) => {
         $vis mod $name {
-            use $crate::src::wrap_fn_ptr::WrappedFnPtr;
             use $crate::src::enum_map::DefaultValue;
             use super::*;
 
-            pub type Fn = WrappedFnPtr<unsafe extern "C" fn($($arg_name: $arg_ty),*) -> $return_ty>;
+            pub type FnPtr = unsafe extern "C" fn($($arg_name: $arg_ty),*) -> $return_ty;
+
+            /// A newtype wrapped [`FnPtr`].
+            ///
+            /// This allows us to add a safer
+            /// (type-safe for sure, and increasingly fully safe)
+            /// wrapper around calling a `fn` ptr.
```

Comment:
```suggestion
            /// method for calling a `fn` ptr.
```

Reply:
This does sound clearer in a way but I meant to imply by wrapper that it's the only public way to call it, whereas method implies more there is also a safer way to call it, but you can still call it the more unsafe way.  If you still think method is clearer though I'll go with it; I don't think it matters too much.

Reply:
I changed it in e885327 to `interface for` instead, as I think that better gets to the root purpose of this type: providing a safer interface, whatever that interface may be.  And I think it implies that this safer interface should shadow/wrap the raw, unsafe one.